CriteoPublisherSdk/Tests/UnitTests/Network/CR_NetworkManagerTests.m

```diff
@@ -172,6 +172,17 @@ - (void)testNetworkManagerPostCall {
                               }]]);
 }
 
+- (void)testNetworkManagerFailedSerialization {
+    NSURL *url;
+    NSData *body;
+    [self.networkManager postToUrl:url
+                              body:body
+                   responseHandler:^(NSData * _Nullable data, NSError * _Nullable error) {
+        XCTAssertNotNil(error);
```

Comment:
If the networkManager is not calling the response handler, then this test could easily pass. Maybe could you add a synchronization point to ensure that the response handler gets invoked ?

See this other test for instance: https://github.com/criteo/ios-publisher-sdk/pull/264/commits/e7a28c15131fa08f885eebd2f776c660412046c8#diff-60683f30ec672a62f5f5605e46eeb74824380a69dc3e3accea852459d77ce1aeR60